### ISSUE TITLE
[core] XMLRenderer: ProcessingErrors from exceptions without a message missing

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1471]
**Affects PMD Version:** it's on the master branch

A ProcessingError could be a `NullPointerException`. If this exception is raised by the JVM, then it usually has no message.

This in turn leads to a NPE in the XMLRenderer, which then doesn't render this and the remaining errors. I think, violations are fully renderer, since they have already been processed.

```
Nov. 16, 2018 9:27:51 NACHM. net.sourceforge.pmd.PMD doPMD
SEVERE: Exception during processing
java.lang.NullPointerException
	at net.sourceforge.pmd.util.StringUtil.appendXmlEscaped(StringUtil.java:214)
	at net.sourceforge.pmd.renderers.XMLRenderer.end(XMLRenderer.java:126)
	at net.sourceforge.pmd.PMD.doPMD(PMD.java:244)
	at net.sourceforge.pmd.PMD.run(PMD.java:485)
	at net.sourceforge.pmd.cli.PMDCommandLineInterface.run(PMDCommandLineInterface.java:165)
	at net.sourceforge.pmd.PMD.main(PMD.java:455)
```

This was the reason for the strange pmdtester report on #1434, where violations have been removed (because a rule failed and was skipped) - but no additional errors have been logged:
http://chunk.io/pmd/6b7d5dbc13504a828f3a8510dd62dfcf/diff/index.html

**Running PMD through:** CLI
